File: cpp/10828.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <sstream>
#include <vector>
#include <stack>
#include <queue>

using std::cin;
using std::cout;

std::stack<int> st;

int main()
{
    cin.tie(NULL);
    cout.tie(NULL);
    std::ios_base::sync_with_stdio(false);

    int ns;
    cin >> ns;

    for (int i = 0; i < ns; i++)
    {
        std::string cmd;
        cin >> cmd;

        if(cmd == "push")
        {
            int n;
            cin >> n;
            st.push(n);
        }
        else if(cmd == "pop")
        {
            if(st.empty())
            {
                cout << -1 << "\n";
                continue;
            }
            
            cout << st.top() << "\n";
            st.pop();
        }
        else if(cmd == "size")
        {
            cout << st.size() << "\n";
        }
        else if(cmd == "empty")
        {
            cout << st.empty() << "\n";
        }
        else if(cmd == "top")
        {
            if(st.empty())
            {
                cout << -1 << "\n";
                continue;
            }
            
            cout << st.top() << "\n";
        }
    }

    return 0;
}

File: python/10828.py
import sys

T = int(sys.stdin.readline().rstrip())
stack = []
for _ in range(T):
    a = list(sys.stdin.readline().rstrip().split())

    if a[0] == 'push':
        stack.insert(0, int(a[1]))
    elif a[0] == 'pop':
        if len(stack) == 0: print(-1)
        else: print(stack.pop(0))
    elif a[0] == 'size':
        print(len(stack))
    elif a[0] == 'empty':
        if len(stack) > 0: print(0)
        else: print(1)
    elif a[0] == 'top':
        if len(stack) == 0: print(-1)
        else: print(stack[0])